data_model/src/account.rs

```diff
@@ -214,15 +226,27 @@ impl Account {
 }
 
 impl NewAccount {
-    fn new(id: AccountId, signatories: impl IntoIterator<Item = PublicKey>) -> Self {
+    fn new(id: AccountId, signature: PublicKey) -> Self {
         Self {
             id,
-            signatories: signatories.into_iter().collect(),
+            signatories: Signatories::from([signature]),
             metadata: Metadata::default(),
         }
     }
 
+    /// Add additional signatures to account.
+    #[inline]
+    #[must_use]
+    pub fn with_additional_signatures(
+        mut self,
+        signatures: impl IntoIterator<Item = PublicKey>,
+    ) -> Self {
+        self.signatories.extend(signatures);
+        self
+    }
```

Comment:
how about:

```suggestion
    pub fn add_signature(
        &mut self,
        signatures: PublicKey,
    ) -> Self {
        self.signatories.extend(signatures);
    }
```

Reply:
let's then do this:

```suggestion
    #[must_use]
    pub fn add_signatory(mut self, signatory: PublicKey) -> Self {
        self.signatories.insert(signatory);
        self
    }
```

we use this pattern also